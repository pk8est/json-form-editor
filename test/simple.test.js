'use strict';

import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';

const packPath = process.env.TEST_LIB ? '../lib/json-editor.min.js' : '../src/JsonEditor.vue';
const pack = require(packPath);
const JsonEditor = pack.default;

const schema = Object.freeze(require('./data/simple.json'));
const model = {
  name: 'Yourtion',
  lists: [ 'Promotion' ],
};
const model2 = {
  name: 'YourtionGuo',
  email: 'yourtion@gmail.com',
};

describe('Component', () => {
  it('Mount', () => {
    const wrapper = shallowMount(JsonEditor, {
      propsData: { schema },
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('Snapshot', () => {
    const renderedString = renderToString(JsonEditor, {
      propsData: { schema },
    });
    expect(renderedString).toMatchSnapshot();
  });

  describe('Mount with data and set data', () => {
    const wrapper = shallowMount(JsonEditor, {
      propsData: { schema, value: model },
    });
    const component = wrapper.vm;
    expect(wrapper.isVueInstance()).toBeTruthy();
    const form = component.$el.getElementsByTagName('form')[0];
    const { name, lists, email } = form.elements;

    it('get mounted data', () => {
      expect(name.getAttribute('value')).toBe(model.name);
      expect(lists.getAttribute('value')).toBe(model.lists[0]);
    });

    it('update value by setData', () => {
      wrapper.setData({ value: model2 });
      expect(email.getAttribute('value')).toBe(model2.email);
      expect(name.getAttribute('value')).toBe(model2.name);
    });
  });

});
