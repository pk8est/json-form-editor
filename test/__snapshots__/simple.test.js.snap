// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshot 1`] = `
<div>
  <h1>Newsletter Subscription</h1>
  <p>Sign up for free newsletters and get more delivered to your inbox</p>
  <form><label placeholder="Your Full Name" title="Please enter your full name" type="text" schemaType="string" label="Full Name" description="" required="required" name="name" minlength="8" maxlength="80"><span data-required-field="true">Full Name</span><input placeholder="Your Full Name" title="Please enter your full name" type="text" value="" schemaType="string" label="Full Name" description="" required="required" name="name" minlength="8" maxlength="80"></label><label type="email" placeholder="Your Email" title="Please enter your email" schemaType="string" label="Email" description="" required="required" name="email" maxlength="120"><span data-required-field="true">Email</span><input type="email" placeholder="Your Email" title="Please enter your email" value="" schemaType="string" label="Email" description="" required="required" name="email" maxlength="120"></label><label placeholder="Select your list subscription" title="Please select your list subscription" schemaType="string" label="List" description="" required="required" name="lists" items="[object Object],[object Object]" type="select"><span data-required-field="true">List</span><select placeholder="Select your list subscription" title="Please select your list subscription" value="" schemaType="string" label="List" description="" required="required" name="lists" items="[object Object],[object Object]" type="select">
        <option value="Daily New">Daily New</option>
        <option value="Promotion">Promotion</option>
      </select></label><label><button type="submit" label="Submit">Submit</button></label></form>
</div>
`;
